src/Core/Content/Media/Path/Contract/Service/MediaSubscriber.php

```diff
@@ -0,0 +1,101 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Content\Media\Path\Contract\Service;
+
+use Shopware\Core\Content\Media\Aggregate\MediaThumbnail\MediaThumbnailDefinition;
+use Shopware\Core\Content\Media\MediaDefinition;
+use Shopware\Core\Content\Media\MediaEntity;
+use Shopware\Core\Content\Media\Path\Implementation\MediaPathUpdater;
+use Shopware\Core\Content\Media\Pathname\UrlGeneratorInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Entity;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityLoadedEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWrittenEvent;
+
+/**
+ * @final This class is not intended to extend, but the event listener calls can be removed via compiler pass
+ */
+class MediaSubscriber
+{
+    public function __construct(
+        private readonly MediaPathUpdater $updater,
+        private readonly AbstractUrlGenerator $generator,
+        private readonly UrlGeneratorInterface $legacyGenerator
+    ) {
+    }
+
+    public function written(EntityWrittenEvent $event): void
+    {
+        if (empty($event->getIds())) {
+            return;
+        }
+        if ($event->getEntityName() === MediaDefinition::ENTITY_NAME) {
+            $this->updater->updateMedia($event->getIds());
```

Comment:
should we filter these ids, to only updating records where the `path` isn't written yet?

Reply:
what would happen if a media gets overwritten? 🤔

Reply:
what would happen if a media gets overwritten? 🤔

Reply:
What do you mean? My conclusion is, that we rewrite that path EACH TIME a media is written. I'm not sure if this is necessary or can lead to issues.

Reply:
I understand your conclusion. I just put another thing to add while checking it.

I think there is just one thing we should check when restricting it to empty paths:
each time a mediaFile is replaced the path must be removed. This can happen when using rename or replace functions in shop.